context/app/static/js/components/genes/Azimuth/Organs.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useMemo } from 'react';
+import OrganTile from 'js/components/organ/OrganTile';
+import Stack from '@mui/material/Stack';
+import Skeleton from '@mui/material/Skeleton';
+
+import { useGeneOrgans } from '../hooks';
+import { useSelectedOrganContext } from './SelectedOrganContext';
+
+function OrganTileSkeleton() {
+  return (
+    <Stack direction="row" gap={4} py={1}>
+      <Skeleton width={225} height={80} />
+      <Skeleton width={225} height={80} />
+      <Skeleton width={225} height={80} />
+      <Skeleton width={225} height={80} />
+    </Stack>
+  );
+}
+
+export default function GeneOrgans() {
+  const { data } = useGeneOrgans();
+  const { selectedOrgan, setSelectedOrgan } = useSelectedOrganContext();
+  const organsWithAzimuth = useMemo(() => {
+    if (!data) return null;
+    return Object.entries(data).reduce(
+      (acc, [name, organ]) => {
+        if (organ.azimuth) {
+          acc[name] = organ;
+        }
+        return acc;
+      },
+      {} as typeof data,
+    );
+  }, [data]);
+
+  useEffect(() => {
+    if (organsWithAzimuth) {
+      setSelectedOrgan(organsWithAzimuth[Object.keys(organsWithAzimuth)[0]]);
+    }
+  }, [organsWithAzimuth, setSelectedOrgan]);
```

Comment:
Forgive me if I'm misunderstanding anything. 

- Could we create the map of organs which have azimuth outside of the component? Does it only depend on the organ yaml files?
- Perhaps it's outside of the scope of this PR, but we don't handle loading state for these components and there is a bit of layout shift. If we did handle loading, could we potentially avoid the `useEffect` to set the selected organ?

Reply:
I have
- Reorganized the gene organs hook to only look up dataset counts for organs with azimuth datasets
- Added fallback skeletons for the reference-based analysis and visualization components
- Removed the useEffect in favor of conditional setState within the render

The last revision was the result of seeing [this pattern recommended](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) in the react docs; as long as a mid-render state update does not cause an infinite loop, calling setState in this manner will make React re-run the render process again before updating the DOM, thus helping to minimize the layout shift.